DQM/Physics/src/SusyDQM.cc

```diff
@@ -0,0 +1,986 @@
+        }
+        if(ooEmooP > eleOneOverEMinusOneOverPCutBarrel) return false;
+        //relative isolation cut (compute isolation with delta beta method)
+        double eleRelIsolation = (ele->pfIsolationVariables().sumChargedHadronPt + max(0.,ele->pfIsolationVariables().sumNeutralHadronEt + ele->pfIsolationVariables().sumPhotonEt - 0.5*ele->pfIsolationVariables().sumPUPt)) / ele->pt();
+        if(eleRelIsolation > eleRelIsoCutBarrel) return false;
+    }
+    else{ //endcap cuts
+        //MaxMissingHits cut
+        if(missHits > eleMaxMissingHitsEndcap) return false;
+        //dEtaIn cut
+        if(fabs(ele->deltaEtaSuperClusterTrackAtVtx()) > eleDEtaInCutEndcap) return false;
+        //dPhiIn cut
+        if(fabs(ele->deltaPhiSuperClusterTrackAtVtx()) > eleDPhiInCutEndcap) return false;
+        //SigmaIetaIeta cut
+        if(ele->full5x5_sigmaIetaIeta() > eleSigmaIetaIetaCutEndcap) return false;
+        //H/E cut
+        if(ele->hcalOverEcal() > eleHoverECutEndcap) return false;
+        //d0 cut
+        if(fabs(-ele->gsfTrack()->dxy(vertex->front().position())) > eleD0CutEndcap) return false;
+        //dZ cut
+        if(fabs(ele->gsfTrack()->dz(vertex->front().position())) > eleDZCutEndcap) return false;
+        //1/E-1/p cut
+        double ooEmooP = 0;
+        if(ele->ecalEnergy() < 1e-6 || !std::isfinite(ele->ecalEnergy())){ //check if electron energy is 0 or infinite
+            ooEmooP = 1e30;
+        }
+        else{
+            ooEmooP = fabs(1.0/ele->ecalEnergy() - ele->eSuperClusterOverP()/ele->ecalEnergy());
+        }
+        if(ooEmooP > eleOneOverEMinusOneOverPCutEndcap) return false;
+        //relative isolation cut (compute isolation with delta beta method)
+        double ele_Isolation = (ele->pfIsolationVariables().sumChargedHadronPt + max(0.,ele->pfIsolationVariables().sumNeutralHadronEt + ele->pfIsolationVariables().sumPhotonEt - 0.5*ele->pfIsolationVariables().sumPUPt));
+        double ele_RelIsolation = ele_Isolation / ele->pt();
+        eleIsolation->Fill(ele_Isolation);
+        if(ele_RelIsolation > eleRelIsoCutEndcap) return false;
+    }
+    return true;
+}
+
+//loose cut-based photon ID
+//NOTE: the isolation values are not quite correct due to an issue with footprint removal!  The ID will not work quite as advertised, but it is a reasonable temporary solution.
+template <typename Mu, typename Ele, typename Pho, typename Jet, typename Met>
+bool SusyDQM<Mu, Ele, Pho, Jet, Met>::goodPhoton(const edm::Event& evt, const Pho* pho) {
+
+    //baseline kinematic cuts
+    if(pho->pt() < phoPtCut) return false;
+    if(fabs(pho->superCluster()->eta()) > phoEtaCut) return false;
+
+    //electron veto -- use the presence of a pixel seed 
+    if(pho->hasPixelSeed()) return false;
+
+    //get photon effective areas for isolation
+    double effAreaChHad = 0;
+    double effAreaNeuHad = 0;
+    double effAreaPhot = 0;
+    if(fabs(pho->superCluster()->eta()) < 1.0){
```

Comment:
change to compute eta once instead of in each if (eg, use a switch statement or cache the variable)